Documentation/design-docs/arm64-intrinsics.md

```diff
@@ -0,0 +1,405 @@
+review separately.  This allows review complexity to be kept manageable.  The O(N^2) nature of the review process will be kept to
+reasonable levels and iterations will be finite.
+
+Every effort will be made to completely elaborate all the methods of the intrinsic `static class` for each review.  This will
+help minimize reopening the same classes for review.
+
+Implementation will be kept separate from from API review.
+
+### Partial implementation of intrinsic `static class`
+
++ `IsSupported` must represents the state of an entire intrinsic `static class`
++ Once API review is complete and approved, it is acceptable to implement approved methods in any order provided tests are added
++ The approved API must be completed before the intrinsic `static class` is included in a release
+
+### Test coverage
+
+As intrinsic support is added test coverage must be extended to provide basic testing
+
+## Half precision floating point
+
+This document will refer to half precision floating point as `Half`.
+
++ Machine learning and Artificial intelligence often use `Half` type to simplify storage and improve processing time.
++ CoreCLR and `CIL` in general do not have general support for a `Half` type
++ There is an open request to expose `Half` intrinsics
++ There is an outstanding proposal to add `System.Half` to support this request
++ Implementation of `Half` features will be adjusted based on
+  + Implementation of the `System.Half` proposal
+  + Availability of supporting hardware (extensions)
+  + General language extensions supporting `Half`
+
+### ARMv8 Half precision support
+
+ARMv8 baseline support for `Half` is limited.  The following operations are supported
+
++ Loads and Stores
++ Conversion to/from `Float`
++ Widening from `Vector128<Half>` to two `Vector128<Float>`
++ Narrowing from two `Vector128<Float>` to `Vector128<Half>`
+
+Recent extension add support for
+
++ General operations on `Half` types
++ Vector operations on `Half` types
+
+### `Half` and ARM64 ABI
+
+Any complete `Half` implementation must conform to the `ARM64 ABI`.
+
+The proposed `System.Half` type must be treated as a floating point type for purposes of the ARM64 ABI
+
+As an argument it must be passed in a floating point register.
+
+As a structure member, it must be treated as a floating point type and enter into the HFA determination logic.
+
+Test cases must be written and conformance must be demonstrated.
+
+## Scalable Vector Extension Support
+
+`SVE`, the Scalable Vector Extension introduces its own complexity.
+
+The extension
++ Creates a new set of SVE registers
++ Each register has a platform specific length
+  + Any multiple of 128 bits
+
+Therefore implementation will not be trivial.
+
++ Register allocator may need changes
++ SIMD support will face similar issues
++ Open issue: Should we use `Vector<T>` or SVE<T> in user interface design?
```

Comment:
Can you articulate the issues for why one might prefer a different type? I realize it's out of scope for this document, but it would be good to indicate why you think it might be desirable.